File: scala/parsers/src/test/scala/jp/co/teruuu/parser/json/JStringParserTest.scala
package jp.co.teruuu.parser.json

import jp.co.teruuu.parser.common.{ParseNg, ParseOk}
import jp.co.teruuu.parser.json.`type`.JString
import org.scalatest.FlatSpec

class JStringParserTest extends FlatSpec{

  "parse" should "successt" in {
    val parser = JParser.string()

    val result1 = parser.parse("\"key\"")
    assert(result1.isInstanceOf[ParseOk[_]])

    val result2 = parser.parse("\"esca\\\"pe\"")
    assert(result2.isInstanceOf[ParseOk[JString]])
    assert(result2.asInstanceOf[ParseOk[JString]].value.equals(JString("esca\"pe")))

    val result3 = parser.parse("key")
    assert(result3.isInstanceOf[ParseNg[_]])
  }
}

File: java/parsers_java8/src/test/java/jp/co/teruuu/parser/json/JStringParserTest.java
package jp.co.teruuu.parser.json;

import jp.co.teruuu.parser.common.ParseResult;
import jp.co.teruuu.parser.json.type.JString;
import org.junit.Test;

public class JStringParserTest {

    @Test
    public void test() {
        JStringParser parser = new JStringParser();

        ParseResult result1 = parser.parse("\"abcdefg\"");
        assert(result1 instanceof ParseResult.Success);
        assert(((ParseResult.Success) result1).value.equals(new JString("abcdefg")));

        ParseResult result2 = parser.parse("\"abc\\\"defg\"");
        assert(result2 instanceof ParseResult.Success);
        assert(((ParseResult.Success) result2).value.equals(new JString("abc\"defg")));
    }
}
